pkg/orchestration/wiring/edge/plugin.go

```diff
@@ -2,52 +2,91 @@ package edge
 )
 
 type WiringPlugin struct {
-	svccatentangler.SvcCatEntangler
 }
 
 func New() *WiringPlugin {
-	return &WiringPlugin{
-		SvcCatEntangler: svccatentangler.SvcCatEntangler{
-			ClusterServiceClassExternalID: ServiceId,
-			ClusterServicePlanExternalID:  PlanId,
-			InstanceSpec:                  getInstanceSpec,
-			//ObjectMeta:                    getObjectMeta,
-			//References:                    getReferences,
-			ResourceType: ResourceType,
+	return &WiringPlugin{}
+}
+
+func (p *WiringPlugin) WireUp(resource *orchestration.StateResource, context *wiring.WiringContext) (result *wiring.WiringResult, retriable bool, err error) {
+	result = nil
+	retriable = false
+
+	if resource.Type != ResourceType {
+		err = errors.Errorf("invalid resource type: %q", resource.Type)
+		return
+	}
+
+	serviceInstance, err := osb.ConstructServiceInstance(resource, ServiceId, PlanId)
+	if err != nil { return }
+
+	instanceParameters, err := instanceParameters(resource, context)
+	if err != nil { return }
+
+	serviceInstance.Spec.Parameters = &runtime.RawExtension{
+		Raw: instanceParameters,
+	}
+
+	serviceInstanceResource := smith.Resource{
+		Name: wiringutil.ServiceInstanceResourceName(resource.Name),
+		Spec: smith.ResourceSpec{
+			Object: serviceInstance,
 		},
 	}
-}
 
-func getInstanceSpec(resource *orchestration.StateResource, context *wiring.WiringContext) (ret []byte, err error) {
-	var params Parameters
-	err = json.Unmarshal(resource.Spec.Raw, &params)
-	if len(params.UpstreamAddress) == 0 {
-		// We need to fill in the upstream from a dependency with DNS
-		// We should also include the region from the context.
-		//
-		// The region acts as a locality tag, so that proxies can match
-		// their own location with the nearest upstream
-		params.UpstreamAddress = UpstreamAddress{
-			{
-				Address: "Get FQDN of dependency somehow",
-				Region:  context.StateContext.Location.Region,
-			},
-		}
-		// ret = <magical json marshalling with 10 other libraries>
+	result = &wiring.WiringResult{
+		Contract: wiring.ResourceContract{
+			Shapes: nil, // Nothing will depend on GlobalEdge
+		},
+		Resources: []smith.Resource{serviceInstanceResource},
 	}
 	return
 }
+
+func instanceParameters(resource *orchestration.StateResource, context *wiring.WiringContext) ([]byte, error) {
+	if resource.Spec == nil {
+		return nil, errors.New("empty spec is not allowed")
+	}
+
+	// Pass-through SD parameters -> OSB attributes
+	var attributes Attributes
+	if err := json.Unmarshal(resource.Spec.Raw, &attributes); err != nil {
+		return nil, errors.WithStack(err)
+	}
+	parameters := InstanceParameters{
+		ServiceName: string(context.StateContext.ServiceName),
+		Resource: ResourceParameters{
+			Attributes: attributes,
+		},
+
+	}
+
+	if parameters.Resource.Attributes.UpstreamAddress == nil {
+		return nil, errors.New("UpstreamAddress is required")
+
+		// TODO: Make upstream address optional and produce it from InternalDNS / KubeIngress output
```

Comment:
This will have to be from EC2Compute and KubeIngress.
NOT InternalDNS

Reply:
Updated